src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java

```diff
@@ -304,7 +321,56 @@ private JPanel getUninstalledAddOnsPanel() {
 		}
 		return uninstalledAddOnsPanel;
 	}
-	
+
+	private static JPanel createFilterPanel(final JXTable table) {
+		JPanel filterPanel = new JPanel();
+		filterPanel.setLayout(new GridBagLayout());
+
+		JLabel filterLabel = new JLabel(Constant.messages.getString("cfu.label.addons.filter"));
+		final JTextField filterTextField = new JTextField();
+
+		filterLabel.setLabelFor(filterTextField);
+		filterPanel.add(filterLabel, LayoutHelper.getGBC(0, 0, 1, 0.0D));
+		filterPanel.add(filterTextField, LayoutHelper.getGBC(1, 0, 1, 1.0D));
+
+		String tooltipText = Constant.messages.getString("cfu.label.addons.filter.tooltip");
+		filterLabel.setToolTipText(tooltipText);
+		filterTextField.setToolTipText(tooltipText);
+
+		// Set filter listener
+		filterTextField.getDocument().addDocumentListener(new DocumentListener() {
+			@Override
+			public void insertUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void removeUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void changedUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			public void updateFilter() {
+				String filterText = filterTextField.getText();
+				if (filterText.isEmpty()) {
+					table.setRowFilter(null);

```

Comment:
The foreground colour should also be reset here (otherwise if the whole "wrong regex" is removed the cursor would still be shown as red).